### ISSUE TITLE
diophantine gives wrong solution for -4*x**2+4*x*y-y**2+2*x-3

### ISSUE THREAD

**Issue by sdemarre** [sympy/sympy#11955]
I used the diophantine solver to try to solve the following equation: `(-y**2)+4*x*y+(-4)*x**2+2*x-3 = 0`
The solver returns 2 solutions. I verified the solution by substituting it back into the equation, expecting 0 as result, but it is not the case.

Here is my session:

```
Python console for SymPy 1.0 (Python 2.7.5)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)

Documentation can be found at http://docs.sympy.org/1.0.

>>> from sympy.solvers.diophantine import diophantine
>>> eq = -4*x**2+4*x*y-y**2+2*x-3
>>> sol = list(diophantine(eq))
>>> eq.subs({x:sol[0][0],y:sol[0][1]}).simplify()
−4t

```
The second solution does seem to be correct:
```
>>> eq.subs({x:sol[1][0],y:sol[1][1]}).simplify()
0

```

**Comment by Abdullahjavednesar** [comment]
@sdemarre are you working on it? I'd like to try this.

**Comment by sdemarre** [comment]
I'm not working on it.
I've been using the sympy diophantine solver to help me implement a solver
for Maxima and accidentally stumbled on this one.

**Comment by ShivramIITG** [comment]
Hello i am newbie .... is it recommended to start contributing with this issue, is anyone working on it already?

**Comment by dhruvkb** [comment]
I have been working on this problem for about nine hours now and have made some valuable progress. It would be nice if I could carry on with this problem and fix it.

**Comment by ShivramIITG** [comment]
sure .. no probs :)

**Comment by Abdullahjavednesar** [comment]
@dhruvkb if you have started off you may go ahead.

**Comment by dharasty** [comment]
I also posted an issue today regarding Diophantine solver not working for elliptical cases: https://github.com/sympy/sympy/issues/11959

Let me know if you need test vectors; I can generate as many as you want.

**Comment by dhruvkb** [comment]
@dharasty Your issue with Diophantine is different from this one. However your issue should be fixed by PR #11961
